.ci/scripts/setup_sklearn.sh

```diff
@@ -26,10 +26,12 @@ sklearn_version=${1:-main}
 if [ "$sklearn_version" == "main" ]; then
     # remove sklearn version from test requirements file
     sed -i.bak -e "s/scikit-learn==1.3.2/scikit-learn/" requirements-test.txt
+    sed -i.bak -e "s/scikit-learn==1.2.2/scikit-learn/" requirements-test.txt
     # install sklearn build dependencies
     pip install threadpoolctl joblib scipy
     # install sklearn from main branch of git repo
     pip install git+https://github.com/scikit-learn/scikit-learn.git@main
 else
     sed -i.bak -e "s/scikit-learn==1.3.2/scikit-learn==${sklearn_version}.*/" requirements-test.txt
+    sed -i.bak -e "s/scikit-learn==1.2.2/scikit-learn==${sklearn_version}.*/" requirements-test.txt
```

Comment:
should we just replace this for any version? not some particular? 
I mean we will not be updating this in case of next version update in requirements 

We ether in case of main would like to go with just scikit-learn without version specification.

and in case of second we will be just forcing version defined 

i.e. with 1.3.3 version we will get this broken